sdks/go/pkg/beam/core/runtime/exec/combine.go

```diff
@@ -319,7 +319,9 @@ func (n *LiftedCombine) FinishBundle(ctx context.Context) error {
 	// Need to run n.Out.ProcessElement for all the cached precombined KVs, and
 	// then finally Finish bundle as normal.
 	for _, a := range n.cache {
-		n.Out.ProcessElement(ctx, a)
+		if err := n.Out.ProcessElement(ctx, a); err != nil {
+			return n.fail(err)
```

Comment:
the n.fail is to mark that *this* Combine stage failed, not the called one. Returning the error directly without n.fail is OK and intended.

The n.fail call below is a mistake. Could you remove the if wrapping and return the error directly?

Reply:
Done. Changed both to return the error directly.